core/state/snapshot/difflayer.go

```diff
@@ -525,19 +525,20 @@ func (dl *diffLayer) StorageList(accountHash common.Hash) ([]common.Hash, bool)
 		dl.lock.RUnlock()
 		return list, destructed // the cached list can't be nil
 	}
+	storageMap := dl.storageData[accountHash]
 	dl.lock.RUnlock()
 
-	// No old sorted account list exists, generate a new one
```

Comment:
I don't believe we will achieve any performance improvement, and this change will make the code harder to understand. Please revert it.

Reply:
Thanks for the explanation, and reverted. :)